server.go

```diff
@@ -118,45 +118,29 @@ func (s *SockchatServer) webSocket(w http.ResponseWriter, r *http.Request) {
 			continue
 		}
 		conn.SetReadDeadline(time.Now().Add(s.timeoutAuthorized))
+		var req any
 		switch receivedMsg.Action {
-		case CreateAction:
-			c := UnmarshalChannelRequest(receivedMsg.Payload)
-			if c == nil {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-				continue
-			}
-			if err := handler.RequestNewChannel(*c, s.channelStore); err != nil {
-				conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-				continue
-			}
-			conn.WriteSocketMsg(NewSocketMessage("channel_created", ChannelRequest{c.Name}))
-
-		case JoinAction:
-			c := UnmarshalChannelRequest(receivedMsg.Payload)
-			if c == nil {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-				continue
-			}
-			if err := handler.RequestJoinChannel(*c, s.channelStore); err != nil {
-				conn.WriteSocketMsg(NewErrorMessage(err.Error()))
-			}
-		case LeaveAction:
-			if c := UnmarshalChannelRequest(receivedMsg.Payload); c != nil {
-				err = handler.RequestLeaveChannel(*c, s.channelStore)
-			} else {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-			}
+		case CreateAction, JoinAction, LeaveAction:
+			req, err = UnmarshalChannelRequest(receivedMsg.Payload)
 		case SendMessageAction:
-			if m := UnmarshalMessageRequest(receivedMsg.Payload); m != nil {
-				err = handler.RequestSendMessage(*m, s.channelStore)
-			} else {
-				conn.WriteSocketMsg(NewErrorMessage(InvalidRequestEvent))
-			}
+			req, err = UnmarshalMessageRequest(receivedMsg.Payload)
+		case LoginAction:
+			// Login request is handled in authorizeConnection
+			continue
```

Comment:
I suggest getting rid of this empty login action handler and treat repeated authorization as an invalid message. 

Currently the UserHandler is taken out of `ConnectedUsersPool` on every message, which locks `ConnectedUsersPool` for some time. It could be extracted once and be constant for the single websocket connection